GitCommands/UserRepositoryHistory/RemoteRepositoryManager.cs

```diff
@@ -72,16 +72,9 @@ public async Task<IList<Repository>> LoadRecentHistoryAsync()
         {
             await TaskScheduler.Default;
 
-            // BUG: this must be a separate settings
-            // TODO: to be addressed separately
-            int size = AppSettings.RecentRepositoriesHistorySize;
             IReadOnlyList<Repository> history = _repositoryStorage.Load(KeyRemoteHistory);
-            if (history is null)
-            {
-                return Array.Empty<Repository>();
-            }
-
-            return AdjustHistorySize(history, size).ToList();
+            const int remotesCacheLength = 20;
```

Comment:
I often need way more than 20 (I do pull a lot of repos).

Reply:
I have set it back to the default value of the setting which is 30

I'm a little circumspect on the added value of this feature because I never found it useful nor used it... Did you already picked a value from this list to set a new remote? Is it intended to be a cache of "syntax examples"?

Reply:
I often pull forks for repos I manage, and since I have my fingers in many pies this history dropdown loses the data I need at the given moment...

Reply:
But that's the thing I don't get. If I understand well, here we want to add a **new** remote so it will **never** contains this new url. At best you will have a url from the same organization/user/git hosting service but never more, right!?!